keps/sig-node/20190212-ephemeral-containers.md

```diff
@@ -358,29 +365,119 @@ There are no limits on the number of Ephemeral Containers that can be created in
+The intended use of this `PodCondition` is to enable administrators to enforce
+custom policies for pods that have had Ephemeral Containers. For example,
+cluster administrators may want to automatically apply a label or delete the pod
+after a configurable time. This may be accomplished by a controller watching
+for this `PodCondition`, though the implementation of such a controller is out
+of scope for this proposal.
+
+### Reattaching and Restarting Ephemeral Containers
 
 One can reattach to a Ephemeral Container using `kubectl attach`. When supported
 by a runtime, multiple clients can attach to a single debug container and share
 the terminal. This is supported by Docker.
 
-### Killing Ephemeral Containers
-
-Ephemeral Containers will not be killed automatically unless the pod is
-destroyed.  Ephemeral Containers will stop when their command exits, such as
-exiting a shell.  Unlike `kubectl exec`, processes in Ephemeral Containers will
-not receive an EOF if their connection is interrupted.
-
-A future improvement could allow killing Ephemeral Containers when they're
-removed from `EphemeralContainers`, but it's not clear that we want to allow
-this. Removing an Ephemeral Container spec makes it unavailable for future
-authorization decisions (e.g. whether to authorize exec in a pod that had a
-privileged Ephemeral Container).
+Ephemeral Containers will not be restarted automatically, and there is no
+method in the API to restart an Ephemeral Container.  Creators of Ephemeral
+Containers are expected to choose a new, unused name.
+
+### Configurable Security Policy
+
+The ability to add containers to a pod implies security trade offs. We've
+received the following requirements and feedback on the alpha implementation:
+
+1.  Admission controllers should be able to enforce policy based on the
+    cumulative pod specification, so operations that prune information,
+    such as removing Ephemeral Containers should not be allowed.
+1.  Restarting a pod is disruptive, so for reasons of operation, security,
+    and resource hygiene it should be possible to delete Ephemeral Containers
+    via the API. ([#84764]).
+1.  Ephemeral Containers could allow privilege escalation greater than that
+    of the initial pod, so setting a custom security context should not be
+    allowed.
+1.  Ephemeral Containers, which are initiated by humans for debugging purposes,
+    should be allowed a more permissive security context regular containers.
+    ([#53188])
+
+Rather than choosing a security posture, policy for Ephemeral Containers
+will be configurable by cluster administrators via 
+[`PodSecurityPolicy`][psp-docs].
+
+[#53188]: https://issues.k8s.io/53188
+[#84764]: https://issues.k8s.io/84764#issuecomment-602004629
+[psp-docs]: https://kubernetes.io/docs/concepts/policy/pod-security-policy/
+
+#### Removing Ephemeral Containers
+
+- Default when `PodSecurityPolicy` controller enabled: disallowed
```

Comment:
PodSecurityPolicy is likely going away, so I don't want to train to closely on that. All of these policy decisions should be enforceable through a general admission controller though.

Reply:
ah ha! well that changes things 😃